app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,51 @@
+class AddressesController < ApplicationController
+    def index 
+        address = Address.all 
+        render json: address 
+    end
+
+    def show
+      id = params[:id]
+      address = Address.find(id)
+      render json: address
+    end
+
+    def new 
+        address = Address.new 
+        render json: address
+    end
+  
+    def create
+      # param keys may subject to change depending on the body of the post request
+      address = Address.create(address_params)
+      if address.valid?
+        # message can be altered for better UI experience in the future
```

Comment:
This is where the address should be created because it passed validations from the Address model.